### ISSUE TITLE
userIP function should be deprecated

### ISSUE THREAD

**Issue by SamMousa** [kartik-v/yii2-helpers#44]
For a while now, Yii2 supports getting the users' IP from request headers in a secure manner.

The current implementation of `Enum::userIP()` is not secure (since it blindly trusts headers) and therefore should not be used.

Furthermore it directly uses the global `$_SERVER` variable which is is considered bad practice.

If a custom implementation is needed (for filtering local IPs), in my opinion it should at the very least use the Yii2 request object.
Note that in most cases filtering local IPs isn't needed if the request object has been properly configured to trust the correct headers.

**Comment by kartik-v** [comment]
Thanks... Could you submit a PR for this?

**Comment by SamMousa** [comment]
Ok, will put it on my list.

**Comment by SamMousa** [comment]
Without the filtering? IE just forward it like this:
````
public static function userIP() {
    return \Yii::$app->request->userIP;
}
````

Or do you want to keep the filtering?

**Comment by kartik-v** [comment]
Resolved via [update to release v1.3.7](https://github.com/kartik-v/yii2-helpers/commit/48ded4252cbf406c39cce49550cab3bb6f317a8f).
